### DISCUSSION TITLE
How to do something before closing a widget to prevent the widget close

### DISCUSSION THREAD

**Discussion by co-index** [eclipse-theia/theia#11865] [Q&A] [answered]
I want to judge whether a widget needs to be closed according to some situations(by clicking the “×” on the window).
<img width="562" alt="image" src="https://user-images.githubusercontent.com/89052431/201891620-df1a0681-96e1-4564-ac2c-0c325e6a02d9.png">

I tried to override the onCloseRequest method in myWidget :
![image](https://user-images.githubusercontent.com/89052431/201891019-3bbb5455-1d7a-46b2-acef-570bf2e6a511.png)

It works, but if I click the “×” again, there will be nothing happened. 

Is there any way to achieve my goal?

**Comment by msujew** [comment, marked as answer]
Hey @MooooooonLight,

I'm not too familiar with phosphorJS, but I believe once the close event has been sent, it won't try to send it again. Meaning you're not allowed to prevent closing by overriding `onCloseRequest`. However, I believe you can achieve your goal by overriding the `close` method instead.

**Reply by co-index** [reply]
hi @msujew thansks for replying
I realized my requirements by setting the "this.title.closeable"
